test/suite/test_prepare_hs04.py

```diff
@@ -0,0 +1,217 @@
+        prepare_value = b"ccccc" * 100
+        for j in range (0, self.nsessions):
+            sessions[j] = self.conn.open_session()
+            sessions[j].begin_transaction('isolation=snapshot')
+            cursors[j] = sessions[j].open_cursor(self.uri)
+            # Each session will update many consecutive keys.
+            start = (j * self.nkeys)
+            end = start + self.nkeys
+            for i in range(start, end):
+                cursors[j].set_key(commit_key + ds.key(self.nrows + i))
+                cursors[j].set_value(prepare_value)
+                self.assertEquals(cursors[j].insert(), 0)
+            sessions[j].prepare_transaction('prepare_timestamp=' + timestamp_str(20))
+
+        hs_writes = self.get_stat(stat.conn.cache_write_hs) - hs_writes_start
+
+        # Assert if not writing anything to the history store.
+        self.assertGreaterEqual(hs_writes, 0)
+
+        # Search the key with timestamp 5 and ignore_prepare=false
+        self.search_key_timestamp_and_ignore(ds, 5, "false")
+
+        # Search the key with timestamp 20 and ignore_prepare=true
+        self.search_key_timestamp_and_ignore(ds, 20, "true")
+
+        # Search the key with timestamp 20 and ignore_prepare=false
+        self.search_key_timestamp_and_ignore(ds, 20, "false")
+
+        # Commit the prepared_transactions with timestamp 30 to search below the key
+        # and expect it not to return prepared conflict message.
+        for j in range (0, self.nsessions):
+            sessions[j].commit_transaction(
+                'commit_timestamp=' + timestamp_str(30) + ',durable_timestamp=' + timestamp_str(30))
+
+        #self.session.commit_transaction()
+        self.session.checkpoint()
+
+        # Simulate a crash by copying to a new directory(RESTART).
+        copy_wiredtiger_home(".", "RESTART")
+
+        # Open the new directory.
+        self.conn = self.setUpConnectionOpen("RESTART")
+        self.session = self.setUpSessionOpen(self.conn)
+
+        # After simulating a crash, search for the keys inserted.
+
+        # Search the key with timestamp 5 and ignore_prepare=false
+        self.search_key_timestamp_and_ignore(ds, 5, "false")
+
+        # Search the key with timestamp 20 and ignore_prepare=false.
+        self.search_key_timestamp_and_ignore(ds, 20, "true", True)
+
+        # Search the key with timestamp 20 and ignore_prepare=false.
+        self.search_key_timestamp_and_ignore(ds, 20, "false", True)
+
+        # Search the key with timestamp 30 and ignore_prepare=false.
+        self.search_key_timestamp_and_ignore(ds, 30, "false", True)
+
+        # Search the key with timestamp 30 and ignore_prepare=true.
+        self.search_key_timestamp_and_ignore(ds, 30, "true", True)
+
+        # Close all cursors and sessions, this will cause prepared updates to be
+        # rollback-ed
+        for j in range (0, self.nsessions):
+            cursors[j].close()
+            sessions[j].close()
```

Comment:
At this point, after the rollback, we should check to make sure the values are correct.  Probably would be one call to `search_keys_timestamp_and_ignore` (assuming it is restructured as suggested).

Reply:
Fixed